### ISSUE TITLE
Log4j Patch - Wrong flags for version

### ISSUE THREAD

**Issue by cobaltgit** [itzg/docker-minecraft-server#1274]
### Describe the problem

I'm running a Minecraft 1.16.5 Purpur server and the Log4j JVM flags don't match the official Minecraft instructions  
> 1.12-1.16.5: Download [this file](https://launcher.mojang.com/v1/objects/02937d122c86ce73319ef9975b58896fc1b491d1/log4j2_112-116.xml) to the working directory where your server runs. Then add the following JVM arguments to your startup command line: 
-Dlog4j.configurationFile=log4j2_112-116.xml

However, my JVM flags are the following:
```
java -XX:+UseShenandoahGC -XX:+AlwaysPreTouch -XX:ShenandoahGCHeuristics=compact -XX:ConcGCThreads=4 -XX:+UseLargePages -XX:+DisableExplicitGC -XX:+UseLargePagesInMetaspace -Xmx12G -Xms12G -Dlog4j2.formatMsgNoLookups=true -jar purpur-1.16.5-1171.jar
```
This Log4j configuration flag was actually provided by Minecraft for 1.17 servers

### Container definition

```yml
version: "3"

services:
  mc:
    image: itzg/minecraft-server:java11
    ports:
      - 25565:25565
    environment:
      EULA: "TRUE"
      VERSION: "1.16.5"
      TYPE: "PURPUR"
      MEMORY: "12G"
      JVM_XX_OPTS: "-XX:+UseShenandoahGC -XX:+AlwaysPreTouch -XX:ShenandoahGCHeuristics=compact -XX:ConcGCThreads=4 -XX:+UseLargePages -XX:+DisableExplicitGC"
      TZ: "Europe/London"
      USE_LARGE_PAGES: "true"
      MODE: "creative"
      SPAWN_PROTECTION: "0"
      VIEW_DISTANCE: "8"
      ENABLE_RCON: "true"
      RCON_PASSWORD: "rcon_passwd_redacted"
    network_mode: "host"

    tty: true
    stdin_open: true
    restart: unless-stopped
    volumes:
      # attach a directory relative to the directory containing this compose file
      - /opt/minecraft/cobalt:/data
  backups:
    image: itzg/mc-backup
    environment:
      BACKUP_INTERVAL: "24h"
      RCON_PASSWORD: "rcon_passwd_redacted"
      # instead of network_mode below, could declare RCON_HOST
      # RCON_HOST: mc
    volumes:
    # mount the same volume used by server, but read-only
    - /opt/minecraft/cobalt:/data:ro
    # use a host attached directory so that it in turn can be backed up
    # to external/cloud storage
    - /mnt/betelgeuse/Backups/Minecraft:/backups
    # share network namespace with server to simplify rcon access
    restart: unless-stopped
    network_mode: "host
```

### Container logs

```
Paste logs here
```

**Comment by itzg** [comment]
That's because I never saw/received guidance for non-vanilla mitigations. I know non-vanilla servers already use a different log4j config file so the vanilla one can't be used as is. If you can provide non Mojang, but authoritative documentation then I can incorporate that.

**Comment by cobaltgit** [comment]
Just read up on it, Purpur 1.16.5 builds aren't patched

**Comment by cobaltgit** [comment]
However, there are ways you can patch it
https://purpurmc.org/docs/Log4j/

**Comment by itzg** [comment]
Thanks! That looks like a similar approach needed for vanilla, so that should be fairly easy.

**Comment by itzg** [comment]
Can you test out the image `itzg/minecraft-server:test-1274` ?

**Comment by cobaltgit** [comment]
Did you push this to latest?

**Comment by itzg** [comment]
> Did you push this to latest?

Yes. Note `-Dlog4j.configurationFile=purpur_log4j2_1141-1165.xml` in the following

```
> docker run -it --rm -e EULA=TRUE -e TYPE=PURPUR -e VERSION=1.16.5 -e SETUP_ONLY=true itzg/minecraft-server
[init] Running as uid=1000 gid=1000 with /data as 'drwxrwxr-x 2 1000 1000 4096 Jan  8 00:33 /data'
[init] Resolved version given 1.16.5 into 1.16.5
[init] Resolving type given PURPUR
[init] Downloading Purpur from https://api.purpurmc.org/v2/purpur/1.16.5/1171/download ...
[init] Creating server.properties in /data/server.properties
[init] Disabling whitelist functionality
[init] Setting white-list to 'false' in /data/server.properties
[init] WARNING: whitelist enabled but not enforced. Set ENFORCE_WHITELIST=TRUE or update 'enforce-whitelist' in server.properties to enforce the whitelist.
[init] Setting enable-rcon to 'true' in /data/server.properties
[init] Setting rcon.password to 'minecraft' in /data/server.properties
[init] Setting rcon.port to '25575' in /data/server.properties
[init] Setting motd to 'A Purpur Minecraft Server powered by Docker' in /data/server.properties
[init] Checking for JSON files.
[init] Setting initial memory to 1G and max to 1G
[init] Starting the Minecraft server...
SETUP_ONLY: java -Xmx1G -Xms1G -Dlog4j.configurationFile=purpur_log4j2_1141-1165.xml -jar purpur-1.16.5-1171.jar
```
